src/common/buffer.cc

```diff
@@ -125,10 +128,12 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     if (r < 0)
       return r;
     buf[r] = '\0';
-    size_t size = strict_strtol(buf, 10, &err);
-    if (!err.empty())
-      return -EIO;
-    buffer_max_pipe_size.set(size);
+    errno = 0;
+    size_t size = strtoul(buf, nullptr, 10);
+    if (errno == 0)
```

Comment:
Please point me to documentation stating that errno can be not changed is no characters are parsed.
I basing on: https://linux.die.net/man/3/strtoul :
<quote>
Notes

Since strtoul() can legitimately return 0 or ULONG_MAX (ULLONG_MAX for strtoull()) on both success and failure, the calling program should set errno to 0 before the call, and then determine if an error occurred by checking whether errno has a nonzero value after the call. 
</quote>